zerver/views/registration.py

```diff
@@ -295,11 +297,16 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
             # create an unusable password).
             password = None
 
+        default_language = "en"
```

Comment:
If this is user signup and not realm creation, I think we should set the default value of `default_language` to realm.default_language instead of hardcoding English. `en` can be the default_language if we can't get the value from browser locale during realm creation.